core/services/prom_reporter.go

```diff
@@ -0,0 +1,119 @@
+package services
+
+import (
+	"context"
+	"database/sql"
+
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/store"
+	"github.com/smartcontractkit/chainlink/core/store/models"
+	"go.uber.org/multierr"
+	"gopkg.in/guregu/null.v4"
+)
+
+//go:generate mockery --name PrometheusBackend --output ../internal/mocks/ --case=underscore
+type (
+	promReporter struct {
+		db      *sql.DB
+		backend PrometheusBackend
+	}
+
+	PrometheusBackend interface {
+		SetUnconfirmedTransactions(int64)
+		SetMaxUnconfirmedBlocks(int64)
+	}
+
+	defaultBackend struct{}
+)
+
+var (
+	promUnconfirmedTransactions = promauto.NewGauge(prometheus.GaugeOpts{
```

Comment:
why globals and not fields in the defaultBackend?

Reply:
I guess simply because this is how we do it everywhere else. Do you think it matters?

Reply:
no concrete impact right now, but I think minimizing variable scope to where it's required is a good defensive move

Reply:
Thinking deeper about this, I see your point but I'm not sure it buys us enough to be worth the extra faff